.github/workflows/build_and_test.yml

```diff
@@ -154,15 +155,10 @@ jobs:
           name: gradle-test-report-${{ matrix.os }}
           path: '**/build/reports/'
           retention-days: 1
-      - name: Code coverage report
-        uses: codecov/codecov-action@v3
-        with:
-          fail_ci_if_error: false
-          token: ${{ secrets.CODECOV_TOKEN }}
 
       - name: 'Publish a snapshot to GitHub'
         id: publish-github
-        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref == 'refs/heads/master' }}
+        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref == 'refs/heads/master' && github.repository == 'saveourtool/diktat' }}
```

Comment:
The only problem here is that we publish snapshot to run diktat-snapshot on our own code (to test diktat with diktat snapshop). So also will need to disable task diktat_snapshot.yml

Reply:
Do you mean you published snapshots from your fork?

This won't prevent snapshots from being published for the main repository, though. It only avoids running on forks.

> So also will need to disable task diktat_snapshot.yml

It looks like that workflow has a job to publish the snapshot locally:

https://github.com/saveourtool/diktat/blob/54a803081c98db78adbb21bb20ab185805314000/.github/workflows/diktat_snapshot.yml#L43-L52

But I didn't have the time to inspect what Gradle's task `generateLibsForDiktatSnapshot` does. Disabling `diktat_snapshot.yml` for forks to be consistent with the changes here is quite straightforward, anyway.

Reply:
Giving this a second thought, it makes sense that `diktat_snapshot.yml` does NOT depend on the snapshots published by the `build_and_test` workflow since they can run concurrently.

Am I reading this right?

Reply:
Ah, yes, you are right! We publish LOCAL snapshots in `diktat_snapshot.yml` task, where we check diktat itself with it's snapshots.